packages/lesswrong/server/collections/automatedContentEvaluations/helpers.ts

```diff
@@ -0,0 +1,279 @@
+      model: "gpt-4.1",
+      instructions: rejectionSystemPrompt,
+      input: `
+Given this previous response, go ahead and call the function 'processPost' now:
+${output_text}`,
+      tools,
+      tool_choice: "required",
+    });
+    toolCall = toolResponse.output.find((c) => c.type === "function_call");
+  }
+  if (!toolCall || toolCall.name !== "processPost") {
+    // eslint-disable-next-line no-console
+    console.error("No function call 'processPost' returned");
+    return;
+  }
+  const args = JSON.parse(toolCall.arguments || "{}");
+  return {
+    cot: output_text,
+    decision: args.decision,
+    reasoning: args.reason,
+  };
+};
+
+const NO_LLM_AUTOREJECT_TEMPLATE = "No LLM (autoreject)";
+
+async function rejectPostForLLM(post: DbPost, context: ResolverContext) {
+  const moderationTemplate = await ModerationTemplates.findOne({ name: NO_LLM_AUTOREJECT_TEMPLATE });
+  if (!moderationTemplate) {
+    // eslint-disable-next-line no-console
+    console.error("Moderation template not found");
+    return
+  }
+  await Posts.rawUpdateOne(
+    { _id: post._id },
+    { 
+      $set: { 
+        rejected: true, 
+        rejectedReason: moderationTemplate.contents?.html ?? "" 
+      } 
+    }
+  );
+  // we're deliberate not sending auto-llm-rejections from a human account, 
+  // because we wanna blankface in this context.
+  await sendRejectionPM({post: { ...post, rejectedReason: moderationTemplate.contents?.html ?? "" }, currentUser: null, context});
+}
+
+export async function createAutomatedContentEvaluation(revision: DbRevision, context: ResolverContext) {
+  // we shouldn't be ending up running this on revisions where draft is true (which is for autosaves) but if we did we'd want to return early.
+  if (revision.draft) return;
+
+  const [validatedEvaluation, llmEvaluation] = await Promise.all([
+    getSaplingEvaluation(revision).catch((err) => {
+      // eslint-disable-next-line no-console
+      console.error("Sapling evaluation failed: ", err);
+      captureException(err);
+      return null;
+    }),
+    getLlmEvaluation(revision, context).catch((err) => {
+      // eslint-disable-next-line no-console
+      console.error("LLM evaluation failed: ", err);
+      captureException(err);
+      return null;
+    }),
+  ]);
+
+  if (!validatedEvaluation && !llmEvaluation) {
+    // eslint-disable-next-line no-console
+    console.error("No evaluation returned");
+    return;
+  }
+
+  await AutomatedContentEvaluations.rawInsert({
+    createdAt: new Date(),
+    documentId: revision.documentId,
+    revisionId: revision._id,
+    score: validatedEvaluation?.score,
+    sentenceScores: validatedEvaluation?.sentence_scores,
+    aiChoice: llmEvaluation?.decision,
+    aiReasoning: llmEvaluation?.reasoning,
+    aiCoT: llmEvaluation?.cot,
+  });
+  if (llmEvaluation?.decision === "review" && (validatedEvaluation?.score ?? 0) > .5) {
+    const documentId = revision.documentId;
+    if (!documentId) return
```

Comment:
We should probably pull this check up earlier... if we're dealing with a revision that somehow doesn't have a documentId, I don't think there's any point in doing any of the content evals